### ISSUE TITLE
Assembled genome too large

### ISSUE THREAD

**Issue by zhangkaihui1** [marbl/verkko#171]
The species I study are haploid, 50% repetitive sequences. Coverage cannot be viewed because there is no reference genome. According to your opinion, I used the ONT and hifi data after pollution removal, and the genome size was 160M. What parameters to set or how to get rid of the excess about double the sequence. Second, how do you remove the redundant sequence in plastid genomes (like mitochondria or chloroplast) after removing the redundant sequence? The attachment is assembly. Homopolymer-compresse.noseq.gfa, which is convenient for uploading and changing to txt format

[assembly.homopolymer-compressed.noseq.txt](https://github.com/marbl/verkko/files/11996765/assembly.homopolymer-compressed.noseq.txt)

**Comment by skoren** [comment]
The graph is very fragmented given the small genome size, how much coverage of HiFi and ONT did you have and how many ONT reads >100kb did you have? What exactly is the input sequence here, a single individual? A clonal pool of individual? A non-clonal pool of individuals?

That said, I don't see obvious noise in the graph. The graph indicates the average HiFi coverage is 20x but there is a lot of fluctuation of that, even within the same component where coverages changes from 20x to 100x and back with no obvious repeat structure. We've seen that in some chromosome regions but this seems much more widespread. Perhaps there is contaminating sequence here from another organism or there are multiple individuals with different genomes where some haplotypes are shared and others are not.

**Comment by zhangkaihui1** [comment]
Thank you for your help, 4.1g Hifi data, 26g ONT data, very few reads >100Kb. Since single-cell amplification was performed on the basis of a single sample (sample with only one genome), the amplified products were sequenced by hifi and ont, so reads were concentrated at 7000bp.

| |
zkh15886723100
|
|
***@***.***
|
---- Replied Message ----
| From | Sergey ***@***.***> |
| Date | 07/10/2023 23:07 |
| To | marbl/verkko ***@***.***> |
| Cc | zhangkaihui1 ***@***.***>,
Author ***@***.***> |
| Subject | Re: [marbl/verkko] Assembled genome too large (Issue #171) |

The graph is very fragmented given the small genome size, how much coverage of HiFi and ONT did you have and how many ONT reads >100kb did you have? What exactly is the input sequence here, a single individual? A clonal pool of individual? A non-clonal pool of individuals?

That said, I don't see obvious noise in the graph. The graph indicates the average HiFi coverage is 20x but there is a lot of fluctuation of that, even within the same component where coverages changes from 20x to 100x and back with no obvious repeat structure. We've seen that in some chromosome regions but this seems much more widespread. Perhaps there is contaminating sequence here from another organism or there are multiple individuals with different genomes where some haplotypes are shared and others are not.

**Comment by skoren** [comment]
We've never tried assembly a single-cell amplified sample. I would suspect this is part of the problem with uneven coverage and larger than expected genome size. Verkko assumes relatively uniform coverage and you will not have that with your sample which will prevent simplification of the graph and potentially introduce errors in resolution. I suspect some amplification errors will also produce redundant sequences increasing the genome size. 

So, in short, I don't think verkko is suited to your data type. You could try to pre-process the data to remove potential chimera and try to make the coverage more even with something like khmer: https://khmer.readthedocs.io/en/latest/user/guide.html but I'm not sure how applicable that is to HiFi/ONT data.

**Comment by Dmitry-Antipov** [comment]
I can also suggest to try a short reads assembler tuned for single cell data, like SPAdes.
Although SPAdes was designed to work with Illumina data, it actually can provide somehow reasonable assemblies based on HiFi reads.

**Comment by skoren** [comment]
Issue is due to single-cell amplification, unlikely to be addressed in the near-term since it introduces a lot of noise in the initial graph and resolution.
